pgcli/pgexecute.py

```diff
@@ -233,7 +233,7 @@ def connect(self, database=None, user=None, password=None, host=None,
         }
         new_params.update(kwargs)
         conn_params.update({
-            k: unicode2utf8(v) for k, v in new_params.items() if v is not None
+            k: unicode2utf8(v) for k, v in new_params.items() if v
```

Comment:
I think this was the spot where #978 went wrong:

https://github.com/dbcli/pgcli/pull/978/files#diff-8859be39408e4fbeea82b923784d9e65L227

Previously, we connected to database in 2 different ways:

* if we had a `dsn`, we updated it with `password` and called `psycopg2.connect` with only one parameter: `dsn=xxx`
* otherwise, we called `psycopg2.connect` with the whole set of parameters (`host`, `user`, etc).

I would suggest that we go back to the same logic. If we have `dsn` provided, our `conn_params` should only contain `dsn` (possibly updated with `password`). Otherwise, include all parameters. This logic can be handled by `pgexecute.py`, without the need for the change in `main.py.

What do you think?

Reply:
Sounds good. I'll add a commit with the changes.

Reply:
If the tests still pass, I'll add a couple of tests to the services feature.